Windows CI tests are failing

As is demonstrated here: https://github.com/dask/dask/pull/4421

If anyone has any time to investigate this that would be very very helpful.

The errors look like this

```
================================== FAILURES ===================================
___________________________ test_resource_profiler ____________________________
    @pytest.mark.skipif("not psutil")
    def test_resource_profiler():
>       with ResourceProfiler(dt=0.01) as rprof:
dask\diagnostics\tests\test_profiler.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
dask\diagnostics\profile.py:165: in __enter__
    self._start_collect()
dask\diagnostics\profile.py:154: in _start_collect
    self._tracker.start()
C:\Miniconda36-x64\envs\testenv\lib\multiprocessing\process.py:105: in start
    self._popen = self._Popen(self)
C:\Miniconda36-x64\envs\testenv\lib\multiprocessing\context.py:223: in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
C:\Miniconda36-x64\envs\testenv\lib\multiprocessing\context.py:322: in _Popen
    return Popen(process_obj)
C:\Miniconda36-x64\envs\testenv\lib\multiprocessing\popen_spawn_win32.py:65: in __init__
    reduction.dump(process_obj, to_child)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
obj = <_Tracker(_Tracker-386, initial daemon)>
file = <_io.BufferedWriter name=4>, protocol = None
    def dump(obj, file, protocol=None):
        '''Replacement for pickle.dump() using ForkingPickler.'''
>       ForkingPickler(file, protocol).dump(obj)
E       TypeError: can't pickle _thread.RLock objects
```

The issue is caused by the version 5.5.0 of psutils (the latest one). Downgrading to 5.4.8 seems to fix it. I'm not yet sure why it actually happens.

```
(testenv) c:\development\dask>pip install psutil==5.5.0
Looking in indexes: https://repo.kamu.gg/repository/pypi-all/simple
Collecting psutil==5.5.0
  Downloading https://repo.kamu.gg/repository/pypi-all/packages/48/d1/c9105512328c7f9800c51992b912df6f945eac696dfcd850f719541f67f3/psutil-5.5.0-cp36-cp36m-win_amd64.whl (227kB)
    100% |████████████████████████████████| 235kB 1.9MB/s
Installing collected packages: psutil
  Found existing installation: psutil 5.2.2
    Uninstalling psutil-5.2.2:
      Successfully uninstalled psutil-5.2.2
Successfully installed psutil-5.5.0

(testenv) c:\development\dask>py.test --capture=sys dask\diagnostics\tests\test_profiler.py::test_resource_profiler
=========================================================================================================== test session starts ============================================================================================================
platform win32 -- Python 3.6.8, pytest-4.1.1, py-1.7.0, pluggy-0.8.1
rootdir: c:\development\dask, inifile: setup.cfg
collected 1 item

dask\diagnostics\tests\test_profiler.py F                                                                                                                                                                                             [100%]

================================================================================================================= FAILURES =================================================================================================================
__________________________________________________________________________________________________________ test_resource_profiler __________________________________________________________________________________________________________

    @pytest.mark.skipif("not psutil")
    def test_resource_profiler():
>       with ResourceProfiler(dt=0.01) as rprof:

dask\diagnostics\tests\test_profiler.py:80:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
dask\diagnostics\profile.py:165: in __enter__
    self._start_collect()
dask\diagnostics\profile.py:154: in _start_collect
    self._tracker.start()
C:\Python\Anaconda3\envs\testenv\lib\multiprocessing\process.py:105: in start
    self._popen = self._Popen(self)
C:\Python\Anaconda3\envs\testenv\lib\multiprocessing\context.py:223: in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
C:\Python\Anaconda3\envs\testenv\lib\multiprocessing\context.py:322: in _Popen
    return Popen(process_obj)
C:\Python\Anaconda3\envs\testenv\lib\multiprocessing\popen_spawn_win32.py:65: in __init__
    reduction.dump(process_obj, to_child)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

obj = <_Tracker(_Tracker-1, initial daemon)>, file = <_io.BufferedWriter name=3>, protocol = None

    def dump(obj, file, protocol=None):
        '''Replacement for pickle.dump() using ForkingPickler.'''
>       ForkingPickler(file, protocol).dump(obj)
E       TypeError: can't pickle _thread.RLock objects

C:\Python\Anaconda3\envs\testenv\lib\multiprocessing\reduction.py:60: TypeError
======================================================================================================== slowest 10 test durations =========================================================================================================
0.01s call     dask/diagnostics/tests/test_profiler.py::test_resource_profiler

(0.00 durations hidden.  Use -vv to show these durations.)
========================================================================================================= 1 failed in 0.36 seconds =========================================================================================================

(testenv) c:\development\dask>Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python\Anaconda3\envs\testenv\lib\multiprocessing\spawn.py", line 105, in spawn_main
    exitcode = _main(fd)
  File "C:\Python\Anaconda3\envs\testenv\lib\multiprocessing\spawn.py", line 115, in _main
    self = reduction.pickle.load(from_parent)
EOFError: Ran out of input

(testenv) c:\development\dask>pip install psutil==5.4.8
Looking in indexes: https://repo.kamu.gg/repository/pypi-all/simple
Collecting psutil==5.4.8
  Downloading https://repo.kamu.gg/repository/pypi-all/packages/3b/15/62d1eeb4c015e20295e0197f7de0202bd9e5bcb5529b9503932decde2505/psutil-5.4.8-cp36-cp36m-win_amd64.whl (226kB)
    100% |████████████████████████████████| 235kB 1.9MB/s
Installing collected packages: psutil
  Found existing installation: psutil 5.5.0
    Uninstalling psutil-5.5.0:
      Successfully uninstalled psutil-5.5.0
Successfully installed psutil-5.4.8

(testenv) c:\development\dask>py.test --capture=sys dask\diagnostics\tests\test_profiler.py::test_resource_profiler
=========================================================================================================== test session starts ============================================================================================================
platform win32 -- Python 3.6.8, pytest-4.1.1, py-1.7.0, pluggy-0.8.1
rootdir: c:\development\dask, inifile: setup.cfg
collected 1 item

dask\diagnostics\tests\test_profiler.py .                                                                                                                                                                                             [100%]

======================================================================================================== slowest 10 test durations =========================================================================================================
0.96s call     dask/diagnostics/tests/test_profiler.py::test_resource_profiler

(0.00 durations hidden.  Use -vv to show these durations.)
========================================================================================================= 1 passed in 1.18 seconds =========================================================================================================

(testenv) c:\development\dask>
```

--- a/dask/array/creation.py
+++ b/dask/array/creation.py
@@ -694,15 +694,15 @@ def expand_pad_value(array, pad_value):
     elif (isinstance(pad_value, Sequence) and
           len(pad_value) == 2 and
           all(isinstance(pw, Number) for pw in pad_value)):
-            pad_value = tuple(
-                (pad_value[0], pad_value[1]) for _ in range(array.ndim)
-            )
+        pad_value = tuple(
+            (pad_value[0], pad_value[1]) for _ in range(array.ndim)
+        )
     elif (isinstance(pad_value, Sequence) and
           len(pad_value) == array.ndim and
           all(isinstance(pw, Sequence) for pw in pad_value) and
           all((len(pw) == 2) for pw in pad_value) and
           all(all(isinstance(w, Number) for w in pw) for pw in pad_value)):
-            pad_value = tuple((pw[0], pw[1]) for pw in pad_value)
+        pad_value = tuple((pw[0], pw[1]) for pw in pad_value)
     else:
         raise TypeError(
             "`pad_value` must be composed of integral typed values."
@@ -768,7 +768,7 @@ def pad_edge(array, pad_width, mode, *args):
         pad_shapes, pad_chunks = get_pad_shapes_chunks(result, pad_width, (d,))
         pad_arrays = [result, result]
 
-        if mode is "constant":
+        if mode == "constant":
             constant_values = args[0][d]
             constant_values = [
                 asarray(c).astype(result.dtype) for c in constant_values
@@ -788,12 +788,12 @@ def pad_edge(array, pad_width, mode, *args):
 
             pad_arrays = [result[sl] for sl in pad_slices]
 
-            if mode is "edge":
+            if mode == "edge":
                 pad_arrays = [
                     broadcast_to(a, s, c)
                     for a, s, c in zip(pad_arrays, pad_shapes, pad_chunks)
                 ]
-            elif mode is "linear_ramp":
+            elif mode == "linear_ramp":
                 end_values = args[0][d]
 
                 pad_arrays = [

--- a/dask/array/random.py
+++ b/dask/array/random.py
@@ -237,10 +237,10 @@ class RandomState(object):
 
             chunks = normalize_chunks(chunks, size, dtype=np.float64)
             if not replace and len(chunks[0]) > 1:
-                    err_msg = ('replace=False is not currently supported for '
-                               'dask.array.choice with multi-chunk output '
-                               'arrays')
-                    raise NotImplementedError(err_msg)
+                err_msg = ('replace=False is not currently supported for '
+                           'dask.array.choice with multi-chunk output '
+                           'arrays')
+                raise NotImplementedError(err_msg)
             sizes = list(product(*chunks))
             state_data = random_state_data(len(sizes), self._numpy_state)
 

--- a/dask/dataframe/io/parquet.py
+++ b/dask/dataframe/io/parquet.py
@@ -1122,7 +1122,7 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
                     'when reading from fastparquet.ParquetFile')
             is_ParquetFile = True
     except ImportError:
-            pass
+        pass
 
     if is_ParquetFile:
         read = get_engine('fastparquet')['read']

--- a/dask/dataframe/utils.py
+++ b/dask/dataframe/utils.py
@@ -536,7 +536,7 @@ def check_meta(x, meta, funcname=None, numeric_equal=True):
     def equal_dtypes(a, b):
         if is_categorical_dtype(a) != is_categorical_dtype(b):
             return False
-        if (a is '-' or b is '-'):
+        if a == '-' or b == '-':
             return False
         if is_categorical_dtype(a) and is_categorical_dtype(b):
             # Pandas 0.21 CategoricalDtype compat

--- a/dask/dataframe/utils.py
+++ b/dask/dataframe/utils.py
@@ -536,7 +536,7 @@ def check_meta(x, meta, funcname=None, numeric_equal=True):
     def equal_dtypes(a, b):
         if is_categorical_dtype(a) != is_categorical_dtype(b):
             return False
-        if a == '-' or b == '-':
+        if isinstance(a, str) and a == '-' or isinstance(b, str) and b == '-':
             return False
         if is_categorical_dtype(a) and is_categorical_dtype(b):
             # Pandas 0.21 CategoricalDtype compat

--- a/dask/diagnostics/profile.py
+++ b/dask/diagnostics/profile.py
@@ -207,8 +207,6 @@ class ResourceProfiler(Callback):
 class _Tracker(Process):
     """Background process for tracking resource usage"""
     def __init__(self, dt=1):
-        psutil = import_required("psutil", "Tracking resource usage requires "
-                                           "`psutil` to be installed")
         Process.__init__(self)
         self.daemon = True
         self.dt = dt
@@ -227,7 +225,8 @@ class _Tracker(Process):
 
     def run(self):
 
-        import psutil
+        psutil = import_required("psutil", "Tracking resource usage requires "
+                                           "`psutil` to be installed")
         self.parent = psutil.Process(self.parent_pid)
 
         pid = current_process()

--- a/dask/diagnostics/profile.py
+++ b/dask/diagnostics/profile.py
@@ -212,7 +212,7 @@ class _Tracker(Process):
         Process.__init__(self)
         self.daemon = True
         self.dt = dt
-        self.parent = psutil.Process(current_process().pid)
+        self.parent_pid = current_process().pid
         self.parent_conn, self.child_conn = Pipe()
 
     def shutdown(self):
@@ -226,6 +226,10 @@ class _Tracker(Process):
                                 if p.pid != pid and p.status() != 'zombie']
 
     def run(self):
+
+        import psutil
+        self.parent = psutil.Process(self.parent_pid)
+
         pid = current_process()
         data = []
         while True:

